src/OpenTelemetrySDK/OpenTelemetryController.cs

```diff
@@ -0,0 +1,152 @@
+
+            return _isOpenTelemetryCapable.Value;
+        }
+
+        public static bool isOpenTelemetryEnvironmentEnabled()
+        {
+            if (_isOpenTelmetryEnvironmentEnabled.HasValue)
+            {
+                return _isOpenTelmetryEnvironmentEnabled.Value;
+            }
+
+            string isOpenTelemetryWorkerLogEnabled = Environment.GetEnvironmentVariable(OTelEnvironmentVariableName);
+            _isOpenTelmetryEnvironmentEnabled = !string.IsNullOrEmpty(isOpenTelemetryWorkerLogEnabled);
+
+            return _isOpenTelmetryEnvironmentEnabled.Value;
+        }
+
+        public bool isOpenTelemetryModuleLoaded()
+        {
+            if (_isOpenTelemetryModuleLoaded.HasValue)
+            {
+                return _isOpenTelemetryModuleLoaded.Value;
+            }
+
+            // Search for the OpenTelemetry SDK in the current session
+            var matchingModules = _pwsh.AddCommand(Utils.GetModuleCmdletInfo)
+                .AddParameter("FullyQualifiedName", Utils.OpenTelemetrySdkName)
+                .InvokeAndClearCommands<PSModuleInfo>();
+
+            // If we get at least one result, we know the OpenTelemetry SDK was imported
+            var numCandidates = matchingModules.Count();
+            var isModuleInCurrentSession = numCandidates > 0;
+
+            if (isModuleInCurrentSession)
+            {
+                var candidatesInfo = matchingModules.Select(module => string.Format(
+                    PowerShellWorkerStrings.FoundOpenTelemetrySdkInSession, module.Name, module.Version, module.Path));
+                var otelSDKModuleInfo = string.Join('\n', candidatesInfo);
+
+                if (numCandidates > 1)
+                {
+                    // If there's more than 1 result, there may be runtime conflicts
+                    // warn user of potential conflicts
+                    _logger.Log(isUserOnlyLog: false, LogLevel.Warning, String.Format(
+                        PowerShellWorkerStrings.MultipleExternalSDKsInSession,
+                        numCandidates, Utils.OpenTelemetrySdkName, otelSDKModuleInfo));
+                }
+                else
+                {
+                    // a single SDK is in session. Report its metadata
+                    _logger.Log(isUserOnlyLog: false, LogLevel.Trace, otelSDKModuleInfo);
+                }
+            }
+
+            _isOpenTelemetryModuleLoaded = isModuleInCurrentSession;
+
+
+            return _isOpenTelemetryModuleLoaded.Value;
+        }
+
+        public static void ResetOpenTelemetryModuleStatus()
+        {
+            _isOpenTelemetryCapable = null;
+            _isOpenTelemetryModuleLoaded = null;
+            _isOpenTelmetryEnvironmentEnabled = null;
+        }
+
+        public void AddStartOpenTelemetryInvocationCommand(OpenTelemetryInvocationContext otelContext)
+        {
+            if (!isOpenTelemetryCapable() || !otelContext.isValid())
+            {
+                return;
```

Comment:
!IsOpenTelemetryCapable() is not a problem - this is the default case for apps which are not configured to use OTel. 
!otelContext.isValid() might be worth logging - this ties into the discussion points I left in the other PR, where I am not sure what the scenario might be where the host doesn't pass TraceContext in the invocation. I'm not sure what the contract is there, I'll add some debug logging just in case